### PR TITLE
Set strict types everywhere (except includes)

### PR DESCRIPTION
**Pull request by weeman1337**
Also adds a PHPCS rule for that

- Did set up a new system and clicked through each view
- CI https://chaos.expert/engelsystem/engelsystem/-/pipelines/37066

depends on https://github.com/engelsystem/engelsystem/pull/1056

### CODE REVIEW

File: .phpcs.xml [hunk 1]
```diff
@@ -39,4 +39,10 @@
     <rule ref="SlevomatCodingStandard.Classes.RequireConstructorPropertyPromotion" />
     <rule ref="SlevomatCodingStandard.Commenting.EmptyComment" />
     <rule ref="SlevomatCodingStandard.Commenting.UselessFunctionDocComment" />
+    <rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
+        <exclude-pattern>/includes</exclude-pattern>
+        <properties>
+            <property name="spacesCountAroundEqualsSign" value="0" />
```
Comment (weeman1337) [inline .phpcs.xml:45]: Option set to match current strict type declarations

File: src/Helpers/ConfigureEnvironmentServiceProvider.php [hunk 2]
```diff
@@ -32,7 +34,7 @@ public function register(): void
     protected function setTimeZone(CarbonTimeZone $timeZone): void
     {
         ini_set('date.timezone', (string)$timeZone);
-        date_default_timezone_set($timeZone);
+        date_default_timezone_set((string)$timeZone);
```
Comment (ximex) [inline src/Helpers/ConfigureEnvironmentServiceProvider.php:37]: space between casting and variable

**Review by ximex — CHANGES_REQUESTED** [review]
looks good

File: src/Helpers/ConfigureEnvironmentServiceProvider.php (hunk 2 above)
Comment (weeman1337) [reply src/Helpers/ConfigureEnvironmentServiceProvider.php:37]: I cannot see a space here…

Comment (weeman1337) [reply src/Helpers/ConfigureEnvironmentServiceProvider.php:37]: Oh. I think I got your comment wrong. You mean you want to have a space there. Hard to guess from half a sentence…

Comment (weeman1337) [reply src/Helpers/ConfigureEnvironmentServiceProvider.php:37]: https://github.com/engelsystem/engelsystem/pull/1056

**Comment by MyIgel** [comment]
@weeman1337 this need a rebase, i would do that after #1066 (ShiftEntry) gets merged.

**Comment by MyIgel** [comment]
Hi @weeman1337, this still needs a rebase ;)

File: src/Middleware/LegacyMiddleware.php [hunk 3]
```diff
@@ -62,7 +64,7 @@ public function process(
             /** @var Translator $translator */
             $translator = $this->container->get('translator');
 
-            $page = 404;
+            $page = '404';
```
Comment (MyIgel) [inline src/Middleware/LegacyMiddleware.php:67]: This is just a "workaround" and setting strict typing breaks the beautiful custom error pages like 404 :(

```suggestion
            $page = 404;
```

Comment (MyIgel) [reply src/Middleware/LegacyMiddleware.php:67]: The `renderPage(string $page,` method should be `renderPage(mixed $page,`

**Review by MyIgel — CHANGES_REQUESTED** [review]
I tested some pages and just fond smaller problems but have the feeling that some edge cases (like the 404 page) might still exist.

**Comment by MyIgel** [comment]
Having one commit with the fixes to make it strict compatible and one which adds all strict types would have been great here to simplify reviews.

**Comment by weeman1337** [comment]
> Having one commit with the fixes to make it strict compatible and one which adds all strict types would have been great here to simplify reviews.

:+1: next time

File: src/Middleware/LegacyMiddleware.php (hunk 3 above)
Comment (weeman1337) [reply src/Middleware/LegacyMiddleware.php:67]: uff

**Comment by MyIgel** [comment]
> +1 next time

oof 0o

**Comment by weeman1337** [comment]
Pipeline https://chaos.expert/engelsystem/engelsystem/-/pipelines/37665

**Pull request merged**